opennms-provision/opennms-provision-persistence/src/main/java/org/opennms/netmgt/provision/persist/policies/ScriptPolicy.java

```diff
@@ -121,13 +130,31 @@ public OnmsNode act(final OnmsNode node) {
             if (compiledScript == null) {
                 LOG.warn("No compiled script available for execution.");
             } else {
-                final SimpleBindings globals = new SimpleBindings();
-                globals.put("node", node);
-                globals.put("LOG", LOG);
-                return (OnmsNode) compiledScript.eval(globals);
+                return sessionUtils.withTransaction(() -> {
+                    try {
+                        // When using sessionUtils (programmatic), the object coming from @Transactional (declarative)
+                        // seems to be invalid and vice versa.
+                        // get the node from the Dao and save the node before returning.
+                        OnmsNode dbNode = node;
+                        if(node.getId() > 0) {
+                            dbNode = m_nodeDao.get(node.getId());
+                        }
+                        final SimpleBindings globals = new SimpleBindings();
+                        globals.put("node", dbNode);
+                        globals.put("LOG", LOG);
+                        dbNode = (OnmsNode) compiledScript.eval(globals);
+                        if(dbNode.getId() > 0) {
+                            m_nodeDao.saveOrUpdate(dbNode);
```

Comment:
Saving the entity here goes against the design of provisiond and should only be done later in the chain by a separate task (i.e. doPersistNodeInfo)